inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -154,16 +154,17 @@ def _match_with_invenio_matcher(obj, eng):
     return _match_with_invenio_matcher
 
 
-def was_already_harvested(record):
-    """Return True if the record was already harvested.
+def was_already_harvested_on_legacy(record):
+    """Return True if the record was already harvested on Legacy
 
     We use the following heuristic: if the record belongs to one of the
-    CORE categories then it was probably ingested in some other way.
+    CORE categories then it was probably ingested on Legacy.
     """
-    categories = get_value(record, 'inspire_categories.term', [])
-    for category in categories:
-        if category.lower() in current_app.config.get('INSPIRE_ACCEPTED_CATEGORIES', []):
-            return True
+    arxiv_eprints = get_value(record, 'arxiv_eprints', [])
+    for arxiv_eprint in arxiv_eprints:
+        for category in arxiv_eprint.get('categories', []):
```

Comment:
Proof is in the tests: https://github.com/inspirehep/inspire-next/blob/6a9279b3f250ca0aa796e5d5fea111f9bfc9f2e4/tests/unit/utils/test_utils_record.py#L199-L244